sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -518,8 +520,32 @@ case class DescribeTableCommand(
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
     val result = new ArrayBuffer[Row]
-    val catalog = sparkSession.sessionState.catalog
+    if (query.isDefined) {
+      query.get match {
+        case _: InsertIntoTable | _: InsertIntoDir =>
+          throw new AnalysisException("Describe command is not supported for insert statements.")
+        case u @ Union(children) if children.forall(_.isInstanceOf[InsertIntoTable]) =>
+          throw new AnalysisException("Describe command is not supported for insert statements.")
+        case u: UnresolvedRelation =>
+          describeTable(sparkSession, u.tableIdentifier, partitionSpec, isExtended, result)
```

Comment:
Can you add a test for this match in `describe-query.sql`?

Reply:
@maropu this should have been already covered in describe-table tests. Basically, 
a simple`describe table mytable` will go through it. Thats because its matched by `queryPrimary` in sqlBase.g4. I will add comments.